### DISCUSSION TITLE
Install Directus on DigitalOcean

### DISCUSSION THREAD

**Discussion by Babelfisch** [directus/directus#6233] [Archive]
I follow the instructions on this [page](https://docs.directus.io/guides/installation/digitalocean-app-platform/) to install Directus as App on DigitalOcean. The Postgres database is online and I can connect to the db from my local computer. The .env file is filled with the correct settings (I hope). But while building the app on DigitalOcean I get only this error:

```
Running custom build command: npx directus bootstrap
[2021-06-13T17:19:12.843452995Z] 17:19:12 ✨ Initializing bootstrap...
[2021-06-13T17:19:38.099172805Z] 17:19:38 🚨 Can't connect to the database
[2021-06-13T17:19:38.107362369Z] building: exit status 1
[2021-06-13T17:19:38.111863347Z] ERROR: failed to build: exit status 1
```

I tried also to set the environment variables directly in the App settings but there is ever the same result.

My database section in the .env file looks like this:

```
DB_CLIENT="pg"
DB_HOST="${db-pg-********.DATABASE_URL}"
DB_PORT=25060
DB_DATABASE="directus"
DB_USER="directus"
DB_PASSWORD="********"
DB_SSL__REJECT_UNAUTHORIZED=true
DB_SSL__CA=${db-pg-********.CA_CERT}
```

Any idea whats going wrong?

**Comment by rijkvanzanten** [comment]
I believe 

`DB_HOST="${db-pg-********.DATABASE_URL}"`

is supposed to be

`DB_HOST="${db-pg-********.HOSTNAME}"`

👍🏻

**Reply by Babelfisch** [reply]
No, this wasn't the problem, I already tried the raw url. It doesn't work with the SSL certificate … no idea why. I solved it with `sslmode=no-verify`in the connection string.

`DB_CONNECTION_STRING="postgresql://…/database?sslmode=no-verify`

**Reply by rijkvanzanten** [reply]
> It doesn't work with the SSL certificate … no idea why.

Have you tried the hostname thing? That's how we run it on DO on various instances, which seems to work as expected 🤷🏻

**Reply by Babelfisch** [reply]
I've tried it with the raw hostname:

`DB_HOST="db-pg-*****.db.ondigitalocean.com"`

It doesn't works. The same hostname in `DB_CONNECTION_STRING` works (but only with `?sslmode=no-verify`)

**Reply by rijkvanzanten** [reply]
Try using [the connection variables that digitalocean provides](https://docs.digitalocean.com/products/app-platform/how-to/use-environment-variables/). Make sure to also configure the port, and other env vars when not using the connection string, as that includes all the different flags

**Reply by yellowstripe-io** [reply]
This is due to the DB variables not being available from digital ocean when the apps platform is in a build state. Not sure why the cert doesnt work that way... but...
